impl/ExtractorRegistry.cpp

```diff
@@ -1,89 +1,88 @@
-
-#include "ExtractorIF.h"
-#include "ExtractorDerived.tpp"
-#include "ExtractorForms.tpp"
-#include "ExtractorRegistry.h"
+#include "mavis/Extractor.h"
+#include "mavis/ExtractorRegistry.h"
+#include "impl/forms/ExtractorForms.h"
+#include "impl/forms/ExtractorDerived.h"
 
 namespace mavis {
 
     const ExtractorIF::PtrType &ExtractorRegistry::getExtractor(const std::string &fname)
     {
         static const std::map<std::string, ExtractorIF::PtrType> EXTRACTOR_REGISTRY = {
```

Comment:
Random though that's way outside the scope of this PR: would it be worth seeing if all of these `static const std::map` instances could be replaced with a compile-time map like https://github.com/mapbox/eternal?

Reply:
That would be great. I don't quite understand why `constexpr` can't be used for maps.